docs/demo.md

```diff
@@ -0,0 +1,227 @@
+
+### Setup Kai VSCode IDE plugin
+
+- Follow along the steps listed in [here](https://github.com/konveyor-ecosystem/kai/tree/main/ide) to intergrate Kai IDE plugin with VSCode.
+
+- Before starting the Kai server, Select model `meta-llama/llama-3-70b-instruct` by uncommenting the following block in `kai/config.toml` file
+
+````yaml
+# **IBM served codellama**
+# ```
+ [models]
+   provider = "ChatIBMGenAI"
+
+   [models.args]
+      model_id = "meta-llama/llama-3-70b-instruct"
+      parameters.max_new_tokens = 2048
+# ```
+````
+
+Make sure to comment the default provider, as there can be only one model provider enabled at any given time.
+
+```yaml
+[models]
+provider = "ChatIBMGenAI"
+
+[models.args]
+model_id = "mistralai/mixtral-8x7b-instruct-v01"
+```
+
+If you are interested in using `gpt-3.5-turbo` refer to the section below,
+
+#### Running Kai with GPT-3.5
+
+- Export your OpenAI api key `export OPENAI_API_KEY="<your-openai-api-key>"`
+
+- Before starting the Kai server, Select model `gpt-3.5-turbo` by uncommenting the following block in `kai/config.toml` file
+
+````yaml
+# **OpenAI GPT 3.5**
+# ```
+ [models]
+   provider = "ChatOpenAI"
+
+   [models.args]
+   model = "gpt-3.5-turbo"
+# ```
+````
+
+- Run the server using `make run-server`
+
+#### Run Kai with demo mode
+
+If you don't have access to GEN_AI key, you can run the server in demo mode which will use cached responses
+
+To run the Kai server in demo mode,
+`GENAI_KEY="dummyvalue" DEMO_MODE=true make run-server`
+
+## Step 2: Demo
+
+In this demo, we will explore Kai's capabilities through practical examples within the Coolstore application.
+
+### 2.1 Clone the coolstore app
+
+Let's clone the Coolstore application, which we will be used demo the migration process to Quarkus.
+
+1. To clone the Coolstore demo from its repository -
+
+   ```bash
+   git clone https://github.com/konveyor-ecosystem/coolstore.git
+   ```
+
+2. Next, switch to the branch of the Coolstore app that's been partially migrated -
+
+   ```bash
+   git checkout partial-migration
+   ```
+
+### 2.2 Run Analysis
+
+We will analyze the Coolstore application using the following migration targets to identify potential areas for improvement:
+
+- containterization
+- jakarta-ee
+- jakarta-ee8+
+- jakarata-ee9+
+- quarkus
+
+1. Open VSCode and load coolstore project if it is not already loaded.
+2. Follow steps from 1 through 4 from https://github.com/konveyor-ecosystem/kai/tree/main/ide#usage-summary
+3. Once the analysis is complete you will see incidents listed on the following files
+
+- `src/main/java/com/redhat/coolstore/model/ShoppingCart.java`
+- `src/main/java/com/redhat/coolstore/model/InventoryEntity.java`
+- `src/main/java/com/redhat/coolstore/service/CatalogService.java`
+- `src/main/java/com/redhat/coolstore/service/ShippingService.java`
+- `src/main/java/com/redhat/coolstore/service/ShoppingCartOrderProcessor.java`
+
```

Comment:
I saw the below file was flagged as well with 2 incidents:
src/main/java/com/redhat/coolstore/rest/RestApplication.java

jakarta-jaxrs-to-quarkus-00020

Reply:
@jwmatthews It is already migrated to `Quarkus` but the rules flag that file. Should I add a note mentioning it is not in scope for this demo?

Reply:
IIRC, it also flags `application.properties` file

Reply:
I didn't see application.properties flagged in my run.

It'd be good to add a note that while the below is flagged we are Ok to skip it.
'src/main/java/com/redhat/coolstore/rest/RestApplication.java'

Reply:
added a note here- https://github.com/konveyor-ecosystem/kai/pull/180/files#diff-d9dcfb114408e4c2ab85697b1dc3d17c24e4b0d7320210f1b13dc6d926f2e3feR120